EpsilonDriverControls/Src/DriverControls.c

```diff
@@ -42,15 +57,24 @@ void sendMusicTask(void const* arg)
 {
```

Comment:
In the future instead of having multiple tasks sending CAN I think just having one task would be so much simpler. In that  case you would have absolutely no need for mutexs or petersons or anything like that. There is a tiny bit of complexity getting each send function to go in the correct frequency but simply do:
```cpp
 int counter = 4; // Obviously have clearer names and value defines
 for (;;)
    osDelayUntil(&prevWakeTime, FASTEST_INTERVAL); // in this case 50
   if (counter == 0)
   {
      sendMusic();
      counter = 4;
   }
   sendDriverTask();
}
```
If it is done like this there is no risk of deadlock and very quick and easy to add new tasks;